data/kernels/denoiseprofile.cl

```diff
@@ -211,7 +214,22 @@ denoiseprofile_accu(read_only image2d_t in, global float4* U2, global float* U4,
   const int y = get_global_id(1);
   const int gidx = mad24(y, width, x);
 
-  if(x >= width || y >= height) return;
+  if(q.x<0)
+  {
+    if(x-q.x >= width || x<-q.x) return;
```

Comment:
This looks a bit strange to me. Can you please elaborate on the logic?

Reply:
Yes of course. Thanks for taking time to look at the code.
The idea of all these tests is to prevent accesses to pixels that are outside the limits of the image.
See bellow, there is a memory access at (x,y)+q and another at (x,y)-q
One very important thing is that q.x and q.y can be negative.
Thus, if we have x=3 and q=-5 for instance, x+q will be negative, and thus outside the range of the line we are considering.
The test (x-q.x<width) checks one of the memory bounds for the access (x,y)-q
The test (x<-q.x) is equivalent to (x+q.x<0). It checks one of the bounds for the memory access (x,y)+q

Reply:
All this sounds ok to me. If @upegelow is ok I'll merge.

Reply:
I tested it and it works fine for me.

Reply:
Thanks, will merge then.